src/files/manager.ts

```diff
@@ -0,0 +1,153 @@
+   * const httpClient = new HttpClient('http://localhost:1337/api');
+   * const filesManager = new FilesManager(httpClient);
+   * ```
+   */
+  constructor(httpClient: HttpClient) {
+    this._httpClient = httpClient;
+
+    debug('initialized files manager');
+  }
+
+  /**
+   * Retrieves a list of files based on optional query parameters.
+   *
+   * @param queryParams - Optional parameters to filter or sort the results.
+   * @returns A promise that resolves to an array of file objects.
+   *
+   * @throws {HTTPError} if the HTTP client encounters connection issues, the server is unreachable, or authentication fails.
+   * @throws {Error} if invalid query parameters are provided.
+   *
+   * @example
+   * ```typescript
+   * const filesManager = new FilesManager(httpClient);
+   *
+   * const files = await filesManager.find({
+   *   filters: { mime: { $contains: 'image' } },
+   *   sort: 'name:asc',
+   * });
+   *
+   * console.log(files); // Array of file objects
+   * ```
+   */
+  async find(queryParams?: FileQueryParams): Promise<FileListResponse> {
+    debug('finding files');
+
+    try {
+      // Validate query parameters before making the request
+      if (queryParams) {
+        validateFileQueryParams(queryParams);
+      }
+
+      let url = '/upload/files';
+
+      if (queryParams) {
+        url = URLHelper.appendQueryParams(url, queryParams);
+      }
+
+      const response = await this._httpClient.get(url);
+      const json = await response.json();
+
+      debug('found %o files', Number(json?.length));
+
+      return json;
+    } catch (error) {
+      // Handle validation errors which are already Error instances
+      if (error instanceof Error) {
+        debug('error finding files: %o', error.message);
+        throw error;
+      }
+
+      // Pass through HTTP errors from the HttpClient
+      debug('unexpected error finding files: %o', error);
+      throw error;
+    }
+  }
+
+  /**
+   * Retrieves a single file by its ID.
+   *
+   * @param fileId - The numeric identifier of the file to retrieve.
+   * @returns A promise that resolves to a single file object.
+   *
+   * @throws {HTTPError} if the HTTP client encounters connection issues, the server is unreachable, or authentication fails.
+   * @throws {Error} if the file ID is invalid or the file does not exist.
+   *
+   * @example
+   * ```typescript
+   * const filesManager = new FilesManager(httpClient);
+   *
+   * const file = await filesManager.findOne(1);
+   *
+   * console.log(file); // File object with details
+   * ```
+   */
+  async findOne(fileId: number): Promise<FileResponse> {
+    validateFileId(fileId);
+
+    debug('finding file with ID %o', fileId);
+
+    const url = `/upload/files/${fileId}`;
+
+    try {
+      const response = await this._httpClient.get(url);
+      const json = await response.json();
+
+      // Check if the response is empty or doesn't have expected structure
+      if (!json || typeof json !== 'object') {
```

Comment:
Just wondering why we're validating the returned object here but not for the `find()`?

Do we need to do it at all?